crates/wasmparser/src/validator/component.rs

```diff
@@ -863,28 +854,9 @@ impl ComponentState {
                     let ty = current.check_type_ref(&ty, types, offset)?;
                     current.add_export(name, url, ty, offset, true)?;
                 }
-                crate::InstanceTypeDeclaration::Alias(alias) => match alias {
-                    crate::ComponentAlias::Outer { kind, count, index }
-                        if kind == ComponentOuterAliasKind::CoreType
-                            || kind == ComponentOuterAliasKind::Type =>
-                    {
-                        match kind {
-                            ComponentOuterAliasKind::CoreType => {
-                                Self::alias_core_type(components, count, index, types, offset)?
-                            }
-                            ComponentOuterAliasKind::Type => {
-                                Self::alias_type(components, count, index, types, offset)?
-                            }
-                            _ => unreachable!(),
-                        }
-                    }
-                    _ => {
-                        return Err(BinaryReaderError::new(
-                            "only outer type aliases are allowed in instance type declarations",
-                            offset,
-                        ))
-                    }
-                },
+                crate::InstanceTypeDeclaration::Alias(alias) => {
+                    Self::add_alias(components, alias, types, offset)?;
```

Comment:
Same comment as above re: validation error messages around illegal encodings of aliases in type declarations.